model/task_history.go

```diff
@@ -351,16 +353,9 @@ func (iter *taskHistoryIterator) GetChunk(v *Version, numBefore, numAfter int, i
 	return chunk, nil
 }
 
-func (self *taskHistoryIterator) GetDistinctTestNames(numCommits int) ([]string, error) {
-	session, mdb, err := db.GetGlobalSessionFactory().GetSession()
-	if err != nil {
-		return nil, errors.Wrap(err, "problem getting database session")
-	}
-	defer session.Close()
-
-	session.SetSocketTimeout(time.Minute)
-
-	pipeline := mdb.C(task.Collection).Pipe(
+func (self *taskHistoryIterator) GetDistinctTestNames(ctx context.Context, env evergreen.Environment, numCommits int) ([]string, error) {
+	opts := options.Aggregate().SetBatchSize(0).SetMaxTime(time.Minute)
```

Comment:
I would set this slightly larger than the load balancer timeout (90 secs?) so that we don't accidentally kill something that might have finished